decaylanguage/dec/dec.py

```diff
@@ -714,14 +711,14 @@ def print_decay_modes(
 
             else:
                 fsp_names = get_final_state_particle_names(dm)
-                l.append((get_branching_fraction(dm), "%-50s" % "  ".join(fsp_names)))
+                ls.append((get_branching_fraction(dm), "%-50s" % "  ".join(fsp_names)))
 
-        l.sort(key=operator.itemgetter(0), reverse=(not ascending))
+        ls.sort(key=operator.itemgetter(0), reverse=(not ascending))
 
-        for bf, info in l:
-            print("%12g : %s" % (bf, info))
+        for bf, info in ls:
+            print("{:12g} : {}".format(bf, info))
 
-    def build_decay_chains(self, mother, stable_particles=[]):
+    def build_decay_chains(self, mother, stable_particles=()):
```

Comment:
Never use a mutable structure in an argument default, it's highly bug-prone. (Note: many of these comments are probably to myself, as I wrote quite a few of theses...)

Reply:
Oh, all these functions for building decay chains are my babies, actually, don't worry. I do tend to use lists as lot even when tuples are just fine, and this one is an example. Indeed the user is meant to give a "list" of particles they want to declare as stable for the purpose of playing with all the info parsed.

Reply:
As a good rule, lists, sets, and dicts should never be default arguments. Too easy to modify across calls. So swapping for tuple or frozenset is an easy fix. Dicts are harder, have to use None and check for it.